src/frontend_cache/cache.c

```diff
@@ -1,6 +1,35 @@
 
 #include "../../inc/main.h"
 
+t_cache* create_frontend_cache(t_pagemap* pagemap) {
```

Comment:
@rswhipple I feel like having this function at the top helps orient other developers to the main purpose of this component of our work, which builds a solid foundation for the search functions below. Hope you don't mind me moving your `search_fast_cache` for that reason!